### ISSUE TITLE
[BUG] mlx.core.random.randint always returns the same value

### ISSUE THREAD

**Issue by Redmept1on** [ml-explore/mlx#429]
**Describe the bug**
When I use ''mlx.core.random.randint'' to generate random integers from the given interval, I notice that every time I run test.py, mlx.core.random.randint always returns the same value 

**To Reproduce**

Include code snippet
```python
import mlx.core as mx
a = mx.random.randint(0, 65535, shape=(2, 2), dtype=mx.uint64)
print(a)
```
![image](https://github.com/ml-explore/mlx/assets/84373897/d29f6af5-4542-4be7-81b1-acec61a40fc4)

**Expected behavior**
Each run of the program should output a different random array

**Desktop (please complete the following information):**
 - OS Version: MacOS 14.2.1
 - Version 0.7.0

**Additional context**
The problem still exists when I construct multiple arrays
```python
import mlx.core as mx
a = mx.random.randint(0, 65535, shape=(2, 2), dtype=mx.uint64)
print(a)
b = mx.random.randint(0, 255, shape=(2, 2), dtype=mx.uint8)
print(b)
```
![image](https://github.com/ml-explore/mlx/assets/84373897/487cec6c-d246-41a4-8286-335c713805ac)

**Comment by awni** [comment]
Definitiely. This one should be a quick fix, we have a fixed seed in the global RNG... it probably makes sense to use a seed based on date/time or something that varies enough.

Relevant lines https://github.com/ml-explore/mlx/blob/main/mlx/random.h#L20-L22

**Comment by Redmept1on** [comment]
Thank you very much for your response to the issue I've reported. I'm currently conducting research involving fuzz testing, and the classification of this issue could significantly impact my findings. In this context, could you kindly clarify if this issue could be considered and labeled as a bug? This would help me better understand the scope of issues that are considered bugs.

**Comment by awni** [comment]
This issue is a gray area in terms of calling it a bug. To me it's working as expected (since that is the result we expect given how we set up the RNG), but since it's not expected behavior for most people, calling it a bug is not unreasonable.

**Comment by awni** [comment]
FWIW most issues are not in such a gray area, this one is kind of unusual.

**Comment by Redmept1on** [comment]
Okay, I have gained a deeper understanding through your answer. Thank you for your explanation and support

**Comment by AvikantSrivastava** [comment]
@awni can you take a look this [PR](https://github.com/ml-explore/mlx/pull/457)

**Comment by AvikantSrivastava** [comment]
@Redmept1on does this look okay to you? I have raised a PR for the fix
<img width="415" alt="Screenshot 2024-01-15 at 2 52 29 AM" src="https://github.com/ml-explore/mlx/assets/41926176/1aa19cd2-0961-4e03-853b-6ac3d98faecf">

**Comment by Redmept1on** [comment]
Thank you for your work. After reviewing your PR, I have a question. Does this change only apply to uint64? Or is this method applicable to other data types? @AvikantSrivastava

**Comment by LeonEricsson** [comment]
> Thank you for your work. After reviewing your PR, I have a question. Does this change only apply to uint64? Or is this method applicable to other data types? @AvikantSrivastava

no this change is independent of whatever dtype you want your mx.array to be

**Comment by Redmept1on** [comment]
Thx! I'm very sorry for missing your comment. @LeonEricsson 
The issue seems to have been resolved in mlx v0.10.0. I'm wondering if I can close this issue. @awni

**Comment by awni** [comment]
Yup, sorry I meant to close it, thanks!
